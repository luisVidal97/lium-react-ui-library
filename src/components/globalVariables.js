
const forms = ['contained', 'outlined'];

const heights = {
  large: '50px',
  medium: '40px',
  small: '30px',
}

const borderRadius = {
  strong: '20px',
  normal: '10px',
  none: '0px'
}

const heightsInputText = {
  large: {
    shape: '40px',
    font: '1.2rem'
  },
  medium:  {
    shape: '32px',
    font: '1rem'
  },
  small: {
    shape: '24px',
    font: '.8rem'
  },
}


const colors = {
  primary: {
      normal: '#115293',
      hover: '#0A0A36',
  },
  secondary: {
    normal: '#DC004E',
    hover: '#9A0036',
  },
  danger: {
      normal: '#ff0000',
      hover: '#cc0000',
  },
  warning: {
    normal: '#ffea05',
    hover: '#e2d006'
  },
  success: {
    normal: '#06c920',
    hover: '#019e16'
  },
  dark: {
    normal: '#000000',
    hover: '#686767',
  },
};

const defaultStyle = {
  normal: '#000000',
  hover: '#686767',
}

export { forms, borderRadius, heights, heightsInputText, colors, defaultStyle } ;
