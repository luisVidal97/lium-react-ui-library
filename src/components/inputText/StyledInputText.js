import styled from 'styled-components';

export const StyledInputText = styled.input`
  margin: 5px;
  height: ${ props => props.shapeSize};
  font-size: ${ props => props.fontSize};
  border-radius: ${ props => props.border};
  border: 1px solid #ccc;
  background-color: ${ props => props.backgroundColor};
  pointer-events: ${ props => props.pointerEvent};

  &:hover{
    border-color: ${props => props.color};
  }

  &:focus{
    outline: none !important;
    border-color: ${props => props.color};
    box-shadow: 0 0 5px ${props => props.color};
  }
`
