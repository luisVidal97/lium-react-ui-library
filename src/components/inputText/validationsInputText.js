import {colors, heightComponents, borderRadius, customizeTheme} from '../globalVariables'


export const validationsInputText = ( { disabled: disabledInputText, color, size, border, themes, type} ) =>{

  let colorsComponent  = JSON.parse(JSON.stringify(colors));
  if (themes) {
    colorsComponent = customizeTheme(themes, JSON.parse(JSON.stringify(colors)));
  }

  const shapeSize = size && heightComponents[size]?
  heightComponents[size].shape
  : heightComponents['medium'].shape;

  const fontSize = size && heightComponents[size]?
  heightComponents[size].font
  : heightComponents['medium'].font;

  color = color && colorsComponent[color]?
    colorsComponent[color].normal
    :
    colorsComponent['primary'].normal;

  border = border && borderRadius[border]?
    borderRadius[border]
    :
    borderRadius['normal'];

  let backgroundColor = 'transparent', pointerEvent = 'auto';
  if (disabledInputText === 'true') {
    backgroundColor = '#E0E0E0' ;
    pointerEvent = 'none'
    color = 'transparent';
  }

  type = (type && type ==='text' || type ==='password') ? type: 'text';

  return {
    shapeSize,
    fontSize,
    color,
    border,
    backgroundColor,
    pointerEvent,
    type
  };
}

