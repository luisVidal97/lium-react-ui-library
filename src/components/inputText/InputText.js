import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import { StyledInputText } from './StyledInputText';
import { validationsInputText } from './validationsInputText';

/**
 * Component that display a input which values can be password or text, with a some styles by default.
 * This functional component uses a specific function that check the variables and convert every props in a concrete
 * value to give it certain features. Note that to develop this component, was used Styled
 * Component library to give dinamic styles according 'props'. Please watch the validationsInputText.js
 * file to know more about the validations. The conditional styles are in StyledInputText.js . Both are in
 * the root of this folder.
 * @param {border, size, disabled, color, placeholder, maxLength, type, value, onChange, theme} props
 */

const InputText = (props) => {

  const sources = validationsInputText(props);

  return (
    <Fragment>
      <StyledInputText
        type={sources.type}
        maxLength={props.maxLength}
        placeholder={props.placeholder}
        value={props.value}
        onChange = {props.onChange}
        {...sources}
      />
    </Fragment>
  )
}

InputText.propTypes = {
  border: PropTypes.string,
  size: PropTypes.string,
  disabled: PropTypes.string,
  color: PropTypes.string,
  placeholder: PropTypes.string,
  maxLength: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  theme: PropTypes.shape({
    primary: PropTypes.shape({
      normal: PropTypes.string,
      hover: PropTypes.string,
    }),
    secondary: PropTypes.shape({
      normal: PropTypes.string,
      hover: PropTypes.string,
    }),
    danger: PropTypes.shape({
      normal: PropTypes.string,
      hover: PropTypes.string,
    }),
    warning: PropTypes.shape({
      normal: PropTypes.string,
      hover: PropTypes.string,
    }),
    success: PropTypes.shape({
      normal: PropTypes.string,
      hover: PropTypes.string,
    }),
    dark: PropTypes.shape({
      normal: PropTypes.string,
      hover: PropTypes.string,
    }),
  }),
}

export default InputText;
