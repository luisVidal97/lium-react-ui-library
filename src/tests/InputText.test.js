import React from 'react';
import { shallow } from'enzyme';

import 'jest-styled-components';
import renderer from 'react-test-renderer';

import { heightsInputText, colors, borderRadius } from '../components/globalVariables';
import InputText from '../components/inputText/InputText';
import {StyledInputText} from '../components/inputText/StyledInputText';

describe('Tests over <InputText/>: features and behavior', () => {

  let wrapper, tree;


  test('should show <Button/> successfully', () => {
    wrapper = shallow(<InputText />);
    expect( wrapper ).toMatchSnapshot();
  });

  test(`should show default style values of text input invoking <InputText /> component (without props).
  Default values = height: 32px; font-size: 1rem; border-radius: 10px; background-color: tranparent; pointer-events: auto;
  input[focus]border-color: #115293;  input[focus]box-shadow: 0 0 5px #115293;
  `, () => {
    tree = renderer.create(<InputText />).toJSON();
    expect(tree).toHaveStyleRule('height', heightsInputText.medium.shape);
    expect(tree).toHaveStyleRule('font-size', heightsInputText.medium.font);
    expect(tree).toHaveStyleRule('border-radius', borderRadius.normal);
    expect(tree).toHaveStyleRule('background-color', 'transparent');
    expect(tree).toHaveStyleRule('pointer-events', 'auto');
    expect(tree).toHaveStyleRule('border-color', colors.primary.normal, { modifier: ':focus', });
    expect(tree).toHaveStyleRule('box-shadow', `0 0 5px ${colors.primary.normal}`, { modifier: ':focus', });
  });

  test('should show default style values of text input invoking <InputText /> (with props but wrong values)', () => {
    tree = renderer.create(<InputText disabled='treu' size='sm' color='green' border='24' moove='large'/>).toJSON();
    expect(tree).toHaveStyleRule('height', heightsInputText.medium.shape);
    expect(tree).toHaveStyleRule('font-size', heightsInputText.medium.font);
    expect(tree).toHaveStyleRule('border-radius', borderRadius.normal);
    expect(tree).toHaveStyleRule('background-color', 'transparent');
    expect(tree).toHaveStyleRule('pointer-events', 'auto');
    expect(tree).toHaveStyleRule('border-color', colors.primary.normal, { modifier: ':focus', });
    expect(tree).toHaveStyleRule('box-shadow', `0 0 5px ${colors.primary.normal}`, { modifier: ':focus', });
  });


  test('should show secondary, large and with strong border input text', () => {
    tree = renderer.create(<InputText color='secondary' size='large' border='strong'/>).toJSON();
    expect(tree).toHaveStyleRule('pointer-events', 'auto');
    expect(tree).toHaveStyleRule('border-color', colors.secondary.normal, { modifier: ':focus', });
    expect(tree).toHaveStyleRule('box-shadow', `0 0 5px ${colors.secondary.normal}`, { modifier: ':focus', });
    expect(tree).toHaveStyleRule('height', heightsInputText.large.shape);
    expect(tree).toHaveStyleRule('font-size', heightsInputText.large.font);
    expect(tree).toHaveStyleRule('border-radius', borderRadius.strong);
    expect(tree).toHaveStyleRule('pointer-events', 'auto');
  });


  test('should show disabled input text', () => {
    tree = renderer.create(<InputText disabled="true" size='small' border="none"/>).toJSON();
    expect(tree).toHaveStyleRule('pointer-events', 'none');
    expect(tree).toHaveStyleRule('background-color', '#E0E0E0');
    expect(tree).toHaveStyleRule('border-radius', borderRadius.none);
    expect(tree).toHaveStyleRule('border-color', 'transparent', { modifier: ':hover', });
    expect(tree).toHaveStyleRule('border-color', 'transparent', { modifier: ':focus', });
    expect(tree).toHaveStyleRule('box-shadow', `0 0 5px transparent`, { modifier: ':focus', });
  });

  test('should not execute function if input text is disabled', () => {
    const doSomething = jest.fn();
    wrapper = shallow( <InputText disabled="true" placeholder="Hello world!" value="Hello" onChange={doSomething}/> );
    wrapper.find(StyledInputText).simulate('click');
    expect(doSomething).not.toHaveBeenCalled();
  });


  test('should change the value of the const if input text change', () => {
    const doSomething = jest.fn();
    const ph = 'email';
    wrapper = shallow( <InputText value='' placeholder={ph} onChange={doSomething}/> );
    wrapper.find(StyledInputText).simulate('change');
    expect(doSomething).toHaveBeenCalled();
    expect( wrapper.find(StyledInputText).props().placeholder).toBe(ph);
  });

});
