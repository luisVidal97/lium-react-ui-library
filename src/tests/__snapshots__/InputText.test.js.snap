// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests over <InputText/>: features and behavior should show <InputText/> successfully 1`] = `
<Fragment>
  <styled.input
    backgroundColor="transparent"
    border="10px"
    color="#115293"
    fontSize="1rem"
    pointerEvent="auto"
    shapeSize="32px"
    type="text"
  />
</Fragment>
`;
